packages/core-app-api/src/routing/RouteTracker.tsx

```diff
@@ -38,19 +38,24 @@ const getExtensionContext = (
     const matches = matchRoutes(routes, { pathname });
 
     // Of the matching routes, get the last (e.g. most specific) instance of
-    // the BackstageRouteObject.
-
+    // the BackstageRouteObject that contains a routeRef. Filtering by routeRef
+    // ensures subRouteRefs are aligned to their parent routes' context.
     const routeMatch = matches
       ?.filter(match => match?.route.routeRefs?.size > 0)
       .pop();
-
     const routeObject = routeMatch?.route;
 
     // If there is no route object, then allow inheritance of default context.
     if (!routeObject) {
       return undefined;
     }
 
+    // If the matched route is the root route (no path), and the pathname is
+    // not the path of the homepage, then inherit from the default context.
+    if (routeObject.path === '' && pathname !== '/') {
```

Comment:
Can you double check that this works when hosting on a subpath like `/backstage`?

Reply:
@benjdlambert It do.

![work-on-sub-paths](https://github.com/backstage/backstage/assets/3496491/7c18f33a-1c3c-46ab-8d94-2e997f83625f)